### PR TITLE
cpu/samd5x/can: fix flag handling on receive

### PR DESCRIPTION
**Pull request by kfessel**
### Contribution description

this sets the eff-flag and the rtr-flag on received frames as required by the can standard for reliable handling of can id

this also adds a helping assertion to the send 

### Issues/PRs references

missing handling of eff /eid / xtd flag was identified by @firas-hamdi in #20640

### CODE REVIEW

**Comment by kfessel** [comment]
I am not sure about the ERR_FLAG the matching ESI bit is only used for CAN-FD frames 

https://www.kernel.org/doc/html/latest/networking/can.html also doesn't cover that well

File: cpu/samd5x/periph/can.c [hunk 1]
```diff
@@ -747,7 +755,13 @@ static void _isr(candev_t *candev)
         else {
             DEBUG_PUTS("Received extended CAN frame");
             frame_received.can_id = dev->msg_ram_conf.rx_fifo_1[rx_get_idx].RXF1E_0.bit.ID;
+            frame_received.can_id |= CAN_EFF_FLAG;
+        }
+        if(dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_0.bit.RTR){
```
Comment (firas-hamdi) [inline cpu/samd5x/periph/can.c:760]:
```suggestion
        if(dev->msg_ram_conf.rx_fifo_1[rx_get_idx].RXF1E_0.bit.RTR){
```

Comment (kfessel) [reply cpu/samd5x/periph/can.c]: thank you -- i squashed that right in

File: cpu/samd5x/periph/can.c [hunk 2]
```diff
@@ -716,7 +719,12 @@ static void _isr(candev_t *candev)
         else {
             DEBUG_PUTS("Received extended CAN frame");
             frame_received.can_id = dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_0.bit.ID;
+            frame_received.can_id |= CAN_EFF_FLAG;
         }
+        if(dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_0.bit.RTR){
```
Comment (benpicco) [inline cpu/samd5x/periph/can.c:724]:
```suggestion
        if (dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_0.bit.RTR) {
```

File: cpu/samd5x/periph/can.c [hunk 3]
```diff
@@ -747,7 +755,13 @@ static void _isr(candev_t *candev)
         else {
             DEBUG_PUTS("Received extended CAN frame");
             frame_received.can_id = dev->msg_ram_conf.rx_fifo_1[rx_get_idx].RXF1E_0.bit.ID;
+            frame_received.can_id |= CAN_EFF_FLAG;
+        }
+        if(dev->msg_ram_conf.rx_fifo_1[rx_get_idx].RXF1E_0.bit.RTR){
```
Comment (benpicco) [inline cpu/samd5x/periph/can.c:760]:
```suggestion
        if (dev->msg_ram_conf.rx_fifo_1[rx_get_idx].RXF1E_0.bit.RTR) {
```

File: cpu/samd5x/periph/can.c [hunk 4]
```diff
@@ -747,7 +755,13 @@ static void _isr(candev_t *candev)
...
+            frame_received.can_id |= CAN_RTR_FLAG;
         }
+
```
Comment (benpicco) [inline cpu/samd5x/periph/can.c:763]:
```suggestion
```

File: cpu/samd5x/periph/can.c (hunk 3 above)
Comment (kfessel) [reply cpu/samd5x/periph/can.c:760]: wasn't there a style check for these ?

File: cpu/samd5x/periph/can.c (hunk 4 above)
Comment (kfessel) [reply cpu/samd5x/periph/can.c:763]: removed the double NL

**Review by benpicco — APPROVED** [review]

**Comment by firas-hamdi** [comment]
Looks good

**Review by firas-hamdi — APPROVED** [review]

**Comment by kfessel** [comment]
thank you for reviewing

**Comment by benpicco** [comment]
CI found that this does not compile

**Pull request merged**